contrib/jemalloc/include/jemalloc/internal/rtree.h

```diff
@@ -211,13 +216,23 @@ rtree_leaf_elm_bits_extent_get(uintptr_t bits) {
 }
 
 JEMALLOC_ALWAYS_INLINE szind_t
-rtree_leaf_elm_bits_szind_get(vaddr_t bits) {
+rtree_leaf_elm_bits_szind_get(uintptr_t bits) {
+#    ifdef __CHERI_PURE_CAPABILITY__
+	/* Lowest bit of offset is the boolean flag -> shift by one for szind */
+	return (szind_t)(cheri_getoffset((void*)bits) >> 1);
```

Comment:
In the original jemalloc encoding, the bits above szindex were always assumed to be 0.  Presumably that's true here, too, but either an explicit mask or a comment back-reffing the paragraph I suggested above might be a good idea.  If szind_t is uint8_t or such, the mask should be a no-op, right?

Reply:
I've added an assertion